ranger-discovery-bundle/src/main/java/io/appform/ranger/discovery/bundle/id/IdGenerator.java

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright (c) 2016 Santanu Sinha <santanu.sinha@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.appform.ranger.discovery.bundle.id;
+
+import com.google.common.base.Preconditions;
```

Comment:
We should move the whole id generation and other logic out to another module.. this probably has no dependency on dw, only on zk.

Reply:
This means the constrains etc everything

Reply:
Decision is to move this to a separate project, and use it here instead. Not changing it now. Resolving the thread